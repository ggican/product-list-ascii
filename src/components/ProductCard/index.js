import React from "react";
import PropTypes from "prop-types";

import ProductCardLoading from "./index.loading";
import StyleProductCard from "./index.style";

import convertSenToUsDollar from "../../utils/convertSenToUsDollar";
import convertDateToTimeSince from "../../utils/convertDateToTimeSince";

const ProductCard = ({
    date = "-",
    face = "-",
    price = 0,
    size = "23",
    isAds = false,
}) => {
    return (
        <StyleProductCard size={size}>
            <div className="product__card--top">
                {!isAds ? (
                    <div className="product__card--face">
                        {face !== "-" && face}
                    </div>
                ) : (
                    <div
                        className="product__card--face 
                    product__card--face-ads"
                    >
                        <img src={face} alt="" />
                    </div>
                )}
            </div>

            {!isAds ? (
                <div className="product__card--bottom">
                    <span className="product__card--date">
                        {date !== "-" && convertDateToTimeSince(date)}
                    </span>
                    <span className="product__card--price">
                        {convertSenToUsDollar(price)}
                    </span>
                    <span className="product__card--size">
                        {`Size ${size}`}
                    </span>
                </div>
            ) : (
                <div className="product__card--bottom">
                    <span className="product__card--date">Adverstiment</span>
                </div>
            )}
        </StyleProductCard>
    );
};

ProductCard.Loading = ProductCardLoading;

ProductCard.defaultProps = {
    date: "-",
    face: "-",
    price: 0,
    size: "0",
    isAds: false,
};
ProductCard.propTypes = {
    /**
      `date` for date product card
      */
    date: PropTypes.string,
    /**
      `face` for image product card
      */
    face: PropTypes.string,
    /**
      `isAds` for flag for ads product card
      */
    isAds: PropTypes.bool,
    /**
      `price` for price product card
      */
    price: PropTypes.number,
    /**
      `size` for size of product
      */
    size: PropTypes.string,
};
export default ProductCard;
