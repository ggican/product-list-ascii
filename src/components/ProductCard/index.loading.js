import React from "react";
import Skeleton from "../Skeleton";
import StyleProductCard from "./index.style";

const ProductCardLoading = () => {
    return (
        <StyleProductCard>
            <div className="product__card--top">
                <Skeleton height="100" width="100%"></Skeleton>
            </div>
            <div className="product__card--bottom">
                <div className="product__card--date">
                    <Skeleton height="7px" width="60%"></Skeleton>
                </div>
                <div className="product__card--price">
                    <Skeleton height="7px" width="30%"></Skeleton>
                </div>
                <div className="product__card--size">
                    <Skeleton height="7px" width="40%"></Skeleton>
                </div>
            </div>
        </StyleProductCard>
    );
};

export default ProductCardLoading;
