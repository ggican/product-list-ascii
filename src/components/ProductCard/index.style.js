import styled from "styled-components";

const StyleProductCard = styled.div`
    background-color: rgb(255, 255, 255);
    border-bottom-left-radius: 4px;
    border-bottom-right-radius: 4px;
    display: block;
    height: 100%;
    padding: 0px;
    width: 100%;
    .product__card {
        &--top {
            background: #e4dd79;
            border-top-left-radius: 4px;
            border-top-right-radius: 4px;
            display: flex;
            min-height: 30vh;
            padding: 0;
            width: 100%;
            img {
                border-top-left-radius: 4px;
                border-top-right-radius: 4px;
                display: block;
                height: 100%;
                object-fit: contain;
                width: 100%;
            }
        }

        &--bottom {
            display: flex;
            flex-direction: column;
            padding: 12px;
            width: 100%;
            /* height: 100%; */

            div {
                display: block;
            }
        }

        &--face {
            align-items: center;
            display: flex;
            font-size: ${props => props.size}px;
            justify-content: center;
            padding: 20px;
            width: 100%;
            width: 100%;
            &-ads {
                padding: 0;
            }
        }

        &--date {
            color: rgba(49, 53, 59, 0.96);
            font-size: 0.86rem;
            line-height: 1.5;
            margin-bottom: 4px;
        }
        &--price {
            font-size: 1rem;
            font-weight: bold;
            line-height: 1.43;
            margin-bottom: 4px;
        }
        &--size {
            color: rgba(49, 53, 59, 0.68);
            font-size: 0.71rem;
            font-weight: bold;
            letter-spacing: normal;
            line-height: 1.4;
            margin-bottom: 4px;
        }
    }
`;

export default StyleProductCard;
