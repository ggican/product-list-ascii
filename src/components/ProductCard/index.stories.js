import React from "react";
import ProductCard from "./index";
import defaultImage from "../../static/default-landscape.png";

export default { title: "Components|Product Card", component: ProductCard };

export const ProductCardDefault = () => {
    return (
        <ProductCard
            date="Thu May 28 2020 05:50:51 GMT+0700 (Western Indonesia Time)"
            face="( `Â·Â´ )"
            price={200}
            size="20"
            isAds={false}
        ></ProductCard>
    );
};
export const ProductCardLoading = () => {
    return <ProductCard.Loading></ProductCard.Loading>;
};

export const ProductCardAdds = () => {
    return <ProductCard face={defaultImage} isAds></ProductCard>;
};
