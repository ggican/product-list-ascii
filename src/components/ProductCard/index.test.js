import React from "react";
import ProductCard from "./index";

describe("ProductCard Component", () => {
    it("renders and create snapshot <ProductCard />", () => {
        expect(
            toJson(
                <ProductCard
                    date="Thu May 28 2020 05:50:51 
                    GMT+0700 (Western Indonesia Time)"
                    face="( `·´ )"
                    price={200}
                    size="20"
                    isAds={false}
                ></ProductCard>,
            ),
        ).toMatchSnapshot();
    });
    it("renders shallow is success <ProductCard />", () => {
        expect(
            shallow(
                <ProductCard
                    date="Thu May 28 2020 05:50:51 
                    GMT+0700 (Western Indonesia Time)"
                    face="( `·´ )"
                    price={200}
                    size="20"
                    isAds={false}
                ></ProductCard>,
            ).exists(),
        ).toBe(true);
    });

    it("check face <ProductCard />", () => {
        const component = shallow(
            <ProductCard
                date="Thu May 28 2020 05:50:51 
                GMT+0700 (Western Indonesia Time)"
                face="( `·´ )"
                price={200}
                size="20"
                isAds={false}
            ></ProductCard>,
        );
        expect(component.find(".product__card--face").text()).toEqual(
            "( `·´ )",
        );
    });
    it("check price <ProductCard />", () => {
        const component = shallow(
            <ProductCard
                date="Thu May 28 2020 05:50:51 
                GMT+0700 (Western Indonesia Time)"
                face="( `·´ )"
                price={200}
                size="20"
                isAds={false}
            ></ProductCard>,
        );
        expect(component.find(".product__card--price").text()).toEqual("$0.04");
    });
});
