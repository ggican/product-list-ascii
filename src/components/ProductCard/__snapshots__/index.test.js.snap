// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard Component renders and create snapshot <ProductCard /> 1`] = `undefined`;
